app/src/main/java/com/futurumgame/base/util/FactoryMapping.java

```diff
@@ -0,0 +1,70 @@
+package com.futurumgame.base.util;
+
+import com.futurumgame.base.factories.Factory;
+import com.futurumgame.base.factories.advanced.LimestoneQuarry;
+import com.futurumgame.base.factories.advanced.MortarMixer;
+import com.futurumgame.base.factories.advanced.StoneQuarry;
+import com.futurumgame.base.factories.basic.ClayPit;
+import com.futurumgame.base.factories.basic.DirtPit;
+import com.futurumgame.base.factories.basic.GravelPit;
+import com.futurumgame.base.factories.basic.OilPump;
+import com.futurumgame.base.factories.basic.SandDune;
+import com.futurumgame.base.factories.basic.WaterMill;
+import com.futurumgame.base.factories.ores.CopperMine;
+import com.futurumgame.base.factories.ores.TinMine;
+import com.futurumgame.base.factories.smelteries.CopperSmeltry;
+import com.futurumgame.base.factories.smelteries.TinSmeltry;
+import com.futurumgame.base.interfaces.IFactoryInstancer;
+import com.futurumgame.base.resources.Resource;
+
+import java.util.HashMap;
+
+public class FactoryMapping {
+
+    private static final HashMap<String, IFactoryInstancer> InstancerMapping = getMapping();
+
+    private FactoryMapping() {
+    }
+
+    public static <T extends Resource> Factory<T> instanceFactory(String name, int level) {
+        IFactoryInstancer instancer = InstancerMapping.get(name);
+        return (Factory<T>) (instancer == null ? null : instancer.instance(level));
+    }
+
+    private static HashMap<String, IFactoryInstancer> getMapping() {
+        HashMap<String, IFactoryInstancer> mapping = new HashMap<>();
+        mapping.put(WaterMill.Name, WaterMill::factory);
+        mapping.put(DirtPit.Name, DirtPit::factory);
+        mapping.put(ClayPit.Name, ClayPit::factory);
+        mapping.put(SandDune.Name, SandDune::factory);
+        mapping.put(GravelPit.Name, GravelPit::factory);
+        mapping.put(OilPump.Name, OilPump::factory);//ToDo: implement factories
+        //mapping.put(WoodForest.Name, WoodForest::factory);
```

Comment:
Remove those comments if they aren't used?

Reply:
marker to see what will be implemented